src/lib/components/Textfield.react.js

```diff
@@ -0,0 +1,139 @@
+// @flow
+
+import React, { Component } from 'react';
+
+import { default as MuiTextField } from '@material-ui/core/TextField';
+import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
+import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
+import getMuiTheme from 'material-ui/styles/getMuiTheme';
+
+type Props = {
+  /** If true, the input element will be focused during the first mount */
+  autoFocus?: boolean,
+  /** The classes to be applied to this component. This keys in this object must be valid CSS rule
+   * names, and the values must be strings for the classnames to be assigned to each rule name
+   * Valid rule names are:
+    root
+    formControl
+    focused
+    disabled
+    colorSecondary
+    underline
+    error
+    multiline
+    fullWidth
+    input
+    inputMultiline
+   */
+  classes?: {
+    root?: string,
+    formControl?: string,
+    focused?: string,
+    disabled?: string,
+    colorSecondary?: string,
+    underline?: string,
+    error?: string,
+    multiline?: string,
+    fullWidth?: string,
+    input?: string,
+    inputMultiline?: string,
+  },
+  /** CSS class name of the root element */
+  className?: string,
+  /** 'primary' | 'secondary' */
+  color: string,
+  /** If true, the input element will be disabled. */
+  disabled?: boolean,
+  /** If true, the label will be displayed in an error state */
+  error?: boolean,
+  /** Dash callback to trigger an event handler */
+  fireEvent?: () => void,
+  /** If true, the input will take up the full width of its container */
+  fullWidth?: boolean,
+  /** The helper text content */
+  helperText?: Node,
+  /** Element ID */
+  id?: string,
```

Comment:
Would you make this a required prop? The component will not work with Dash correctly without an `id`, so let's enforce that the user always gives this a value.